src/test/crimson/test_socket.cc

```diff
@@ -0,0 +1,482 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include <seastar/core/app-template.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/sharded.hh>
+#include <seastar/core/sleep.hh>
+
+#include "crimson/common/log.h"
+#include "crimson/net/Errors.h"
+#include "crimson/net/Fwd.h"
+#include "crimson/net/Socket.h"
+
+namespace {
+
+using seastar::future;
+using ceph::net::error;
+using ceph::net::Socket;
+using ceph::net::SocketFRef;
+using ceph::net::stop_t;
+
+seastar::logger& logger() {
+  static auto logger = seastar::logger("test");
```

Comment:
I think we need to register the logger much earlier than on a first call to `logger()`:
```
$ ./bin/unittest_seastar_socket --logger-log-level test=trace
logging configuration error: Unknown logger 'test'. Use --help-loggers to list available loggers.
```

With this dirty fix it works:
```cpp
static auto my_logger = seastar::logger("test");
seastar::logger& logger() {
  return my_logger;
}
```
```
$ ./bin/unittest_seastar_socket --logger-log-level test=trace
WARN  2019-06-25 15:29:41,965 [shard 0] seastar - Unable to set SCHED_FIFO scheduling policy for timer thread; latency impact possible. Try adding CAP_SYS_NICE
INFO  2019-06-25 15:29:42,011 [shard 0] test - test_refused()...
INFO  2019-06-25 15:29:42,011 [shard 0] test - test_bind_same()...
INFO  2019-06-25 15:29:42,011 [shard 0] test - test_accept()
INFO  2019-06-25 15:29:42,112 [shard 0] test - test_read_write()...
INFO  2019-06-25 15:29:42,114 [shard 0] test - test_unexpected_down()...
DEBUG 2019-06-25 15:29:42,117 [shard 1] test - dispatch_read_unbounded(): expected error std::system_error (error ceph::net:4, read eof)
DEBUG 2019-06-25 15:29:42,117 [shard 1] test - dispatch_write_unbounded(): expected error std::system_error (error system:32, sendmsg: Broken pipe)
INFO  2019-06-25 15:29:42,117 [shard 0] test - test_shutdown_propagated()...
DEBUG 2019-06-25 15:29:42,117 [shard 1] test - dispatch_read_unbounded(): expected error std::system_error (error ceph::net:4, read eof)
DEBUG 2019-06-25 15:29:42,117 [shard 1] test - dispatch_write_unbounded(): expected error std::system_error (error system:32, sendmsg: Broken pipe)
INFO  2019-06-25 15:29:42,117 [shard 0] test - test_preemptive_down()...
DEBUG 2019-06-25 15:29:42,220 [shard 0] test - dispatch_write_unbounded(): expected error std::system_error (error system:32, sendmsg: Broken pipe)
DEBUG 2019-06-25 15:29:42,220 [shard 0] test - dispatch_read_unbounded(): expected error std::system_error (error system:104, read: Connection reset by peer)
DEBUG 2019-06-25 15:29:42,220 [shard 1] test - dispatch_write_unbounded(): expected error std::system_error (error system:104, sendmsg: Connection reset by peer)
DEBUG 2019-06-25 15:29:42,220 [shard 1] test - dispatch_read_unbounded(): expected error std::system_error (error ceph::net:4, read eof)
INFO  2019-06-25 15:29:42,220 [shard 0] test - All tests succeeded
```
However, maybe let's use just the `ceph_subsys_ms` instead?
```cpp

seastar::logger& logger() {
  return ceph::get_logger(ceph_subsys_ms);
}
```

Reply:
> However, maybe let's use just the `ceph_subsys_ms` instead?

the previous revision was using `ceph_subsys_ms` . but i was against this . because , in future, if we want to debug the test, the "ms" logger should be different than that of "test". as they belong to different domains and abstraction levels.

i think 
```c++
static logger = seastar::logger{"test"};
```
is good enough.

Reply:
> because , in future, if we want to debug the test, the "ms" logger should be different than that of "test".

Isn't this a generic issue? I believe people working outside crimson may have the same needs in the matter of differentiating between a subsystem and its unit tests. If so, maybe we should consider introduction of a dedicated entry to `common/subsys.h`?

Reply:
on a second thought, why shall we use logging in a unit test? IMHO, for a unittest, no news is good news.

Reply:
> on a second thought, why shall we use logging in a unit test? IMHO, for a unittest, no news is good news.

In an optimistic scenario nobody is really interested in logs.  However, unit tests are just programs still written by humans, so erroneous situations are expendable, I believe. Having logs, operating in similar manner to the component the unit test is made for, seems useful.

Reply:
One reason to add logs is that I found `dispatch_write_unbounded` will throw connection reset and broken pipe randomly. And a similar case is with `dispatch_read_unbounded`. Logging will help us understand the issue here.